cloud/amazon/ec2_vpc_net.py

```diff
@@ -0,0 +1,338 @@
+version_added: "2.0"
+options:
+  name:
+    description:
+      - The name to give your VPC. This is used in combination with the cidr_block paramater to determine if a VPC already exists.
+    required: yes
+    default: null
+    aliases: []
+  cidr_block:
+    description:
+      - The CIDR of the VPC
+    required: yes
+    default: null
+    aliases: []
+  tenancy:
+    description:
+      - Whether to be default or dedicated tenancy. This cannot be changed after the VPC has been created.
+    required: false
+    default: default
+    aliases: []
+  dns_support:
+    description:
+      - Whether to enable AWS DNS support.
+    required: false
+    default: true
+    aliases: []
+  dns_hostnames:
+    description:
+      - Whether to enable AWS hostname support.
+    required: false
+    default: true
+    aliases: []
+  dhcp_id:
+    description:
+      - the id of the DHCP options to use for this vpc
+    default: null
+    required: false
+    aliases: []
+  igw_state:
+    description:
+      - Whether the VPC should have an IGW. If set to true, the module will try to check if any idle IGWs are availble otherwiseit will attempt to create a new igw and attach it to the VPC.
+    default: present
+    required: false
+    aliases: []
+  tags:
+    description:
+      - The tags you want attached to the VPC. This is independent of the name value, note if you pass a 'Name' key it would override the Name of the VPC if it's different.
+    default: null
+    required: false
+    aliases: null
+  state:
+    description:
+      - The state of the VPC. Either absent or present.
+    default: present
+    required: false
+    aliases: null
+  multi_ok:
+    description:
+      - By default the module will not create another VPC if there is another VPC with the same name and CIDR block. Specify this as true if you want duplicate VPCs created.
+    default: false
+    required: false
+    aliases: null
+author: Jonathan Davila
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Create a VPC with dedicate tenancy and a couple of tags
+
+- ec2_vpc:
+    name: Module_dev2
+    cidr_block: 170.10.0.0/16
+    region: us-east-1
+    tags:
+      new_vpc: ec2_vpc_module
+      this: works22
+    igw_state: present
+    tenancy: dedicated
+
+'''
+
+
+import time
+import sys
+
+try:
+    import boto
+    import boto.ec2
+    import boto.vpc
+    from boto.exception import EC2ResponseError
+
+    HAS_BOTO = True
+except ImportError:
+    sys.exit(msg='Boto is required')
+
+
+def vpc_exists(module, vpc, name, cidr_block, multi):
+    """Returns True or False in regards to the existance of a VPC. When supplied
+    with a CIDR, it will check for matching tags to determine if it is a match
+    otherwise it will assume the VPC does not exist and thus return false. """
+    exists=False
+    matched_vpc=None
+    matching_vpcs=vpc.get_all_vpcs(filters={'tag:Name' : name, 'cidr-block' : cidr_block})
```

Comment:
If there is an exception here it would have been raised earlier before this call. When nothing matches it returns an empty list. Def need to do some other captures though.